packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -495,20 +495,20 @@ export class ConnectionService {
         type: message.type,
       })
 
+      const recipientVerkey = messageContext.recipientVerkey && didKeyToVerkey(messageContext.recipientVerkey)
```

Comment:
The more I go through this code, the more I'm reluctant to this approach. Basically we now say a value of `recipientKey`, `recipientVerkey`, `routingKey` or `routingVerkey` can always be either a did:key or a base58 encoded key.

What if we start attaching an instance of `Key` as the `recipientKey` to the message context, so we don't have to do this dance everywhere? It really makes it uncertain which value to expect